src/main/java/org/jenkinsci/plugins/docker/commons/KeyMaterialContext.java

```diff
@@ -0,0 +1,34 @@
+package org.jenkinsci.plugins.docker.commons;
+
+import hudson.FilePath;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+
+/**
+ * Represents the context within a {@link KeyMaterialFactory} can {@link KeyMaterialFactory#materialize()}
+ * {@link KeyMaterial} instances.
+ *
+ * @author Stephen Connolly
+ */
+public class KeyMaterialContext implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    @Nonnull
+    private final FilePath baseDir;
+
+    public KeyMaterialContext(@Nonnull FilePath baseDir) {
+        this.baseDir = baseDir;
+    }
+
+    /**
+     * Returns the base directory that can be used to {@link KeyMaterialFactory#materialize()}
+     * {@link KeyMaterial} instances.
+     *
+     * @return the base directory.
+     */
+    @Nonnull
+    public FilePath getBaseDir() {
+        return baseDir;
+    }
```

Comment:
Well that's a question for @tfennelly as the potential leak there exists before this case... we just highlight it better. My thinking is that basedir should be more like a directory that you can create temp stuff in... so it should always exist and thus not need tidy-up. It's the sub-dirs in that base dir that each key material factory is instantiated into that need to tidy-up, which is what `KeyMaterial.close()` does